pkg/providers/amifamily/bottlerocket.go

```diff
@@ -39,12 +40,14 @@ type Bottlerocket struct {
 }
 
 func (b Bottlerocket) DescribeImageQuery(ctx context.Context, ssmProvider ssm.Provider, k8sVersion string, amiVersion string) (DescribeImageQuery, error) {
+	// Bottlerocket AMIs versions are prefixed with a v on GitHub, but not in the SSM path. We should accept both.
+	trimmedAMIVersion := strings.TrimLeft(amiVersion, "v")
```

Comment:
Are we going to allow this for AL2 and AL2023 as well? I'm not sure I see a reason not to? Why not generalize here to reduce confusion?